### DISCUSSION TITLE
The default position is missing on the category page.

### DISCUSSION THREAD

**Discussion by mahesh-makwana-web-vision** [Smile-SA/elasticsuite#3186] [Q&A]
The default sorting option is 'Position,' and we can change the default sorting option from the category page. Here, we can assign different position numbers to products, as shown below:

![Gear-ID-3-Categories-Inventory-Catalog-Magento-Admin (2)](https://github.com/Smile-SA/elasticsuite/assets/116068098/57bd16ea-9a8e-4b25-9b28-40634e2fdbe1)

After installing the Smile module, we enabled default position sorting from the store view, but we cannot see the position column in the category grid.

![Gear-ID-3-Categories-Inventory-Catalog-Magento-Admin (3)](https://github.com/Smile-SA/elasticsuite/assets/116068098/598cb51d-23a0-4f45-9552-9a811e5cb83b)

**Am I missing something, or is it a bug? Can you please help me solve this problem?**

@romainruaud

**Comment by mahesh-makwana-web-vision** [comment]
Hello @romainruaud

I updated the category position in the catalog_category_product table and performed reindexing and cache cleaning, but the position is still not working.
![database-local-localhost-mz-catalog_category_product-phpMyAdmin-5-2-1](https://github.com/Smile-SA/elasticsuite/assets/116068098/f5251781-7bf7-4f63-9580-2bb1648878c0)

We have more than 10,000 products in one category, so it's not possible to update them manually for position. We have an order number for each product. I checked this link : https://github.com/Smile-SA/elasticsuite/wiki/ProductSorting#manual-product-sorting-in-categories but it didn't work. 

Can you please help me?

**Comment by romainruaud** [comment]
Hi @mahesh-makwana-web-vision 

If you want to import positions in Elasticsuite, you'll want to target the "smile_virtualcategory_catalog_category_product_position" table.

There is no column to change the position of product in admin, you should drag and drop them in the listing, by switching them from "Automatic sort" to "Manual Sort".

The "Use Default position" for a specific store view is just a mechanism to determine if you want to use the same positions at the "Default" level and at the "Store" level. It's not a way to get back the old fashioned grid.

We do not recommend setting a huge amount of products to a pinned position, that's barely unusable for the endusers, it's better to use boosts (optimizers) to create ranking rules.

Where does your positions come from ? Is this something computed from another tool ?

Regards

**Comment by borishinzer** [comment]
Sorry, but the drag'n'drop might be useful for 20-100 products, but not for thousand of products.

**Comment by romainruaud** [comment]
HI @borishinzer that's my point, pinning thounsands of products to fixed positions is not a proper way to do merchandising. Because you'll have permanently to come and change the positions.

> We do not recommend setting a huge amount of products to a pinned position, that's barely unusable for the endusers, it's better to use boosts (optimizers) to create ranking rules.

**Comment by mahesh-makwana-web-vision** [comment]
Hello @romainruaud 

We use MHSI to import products with positions in category pages. You can find MHSI here: [MHSI GitHub Repository](https://github.com/web-vision/magento-high-speed-import).

Customers decide the positions of their products in category listings, so they import products with positions set. 

To enable manual sorting for all products because we cannot change every category manually, we have two options: either modify MHSI or the Smile module.

Is there any possible way to resolve the above issue?

Thanks.

**Comment by romainruaud** [comment]
@mahesh-makwana-web-vision modify your import so that you push the positions into "smile_virtualcategory_catalog_category_product_position" table.

Regards

**Comment by mahesh-makwana-web-vision** [comment]
![Gear-ID-3-Categories-Inventory-Catalog-Magento-Admin (4)](https://github.com/Smile-SA/elasticsuite/assets/116068098/12d3c53f-487d-46a1-947c-ac3554590223)

Is there any global setting to update sorting in all categories without manual work?

**Comment by mahesh-makwana-web-vision** [comment]
Hello @romainruaud 

Is there any global setting to update sorting in all categories without manual work?

Thanks.
